cartographer_rviz/cartographer_rviz/submaps_display.cc

```diff
@@ -149,6 +162,53 @@ void SubmapsDisplay::update(const float wall_dt, const float ros_dt) {
   }
 }
 
+void SubmapsDisplay::AllEnabledChanged() {
+  ::cartographer::common::MutexLocker locker(&mutex_);
+  bool visibility = visibility_all_enabled_->getBool();
+  for (auto &trajectory : trajectories_) {
+    for (auto &submap : trajectory) {
+      DisconnectVisibilityChange(submap);
+      submap->SetVisibility(visibility);
+      ConnectVisibilityChange(submap);
+    }
+  }
+}
+
+void SubmapsDisplay::VisibilityChanged(DrawableSubmap* submap) {
```

Comment:
Are you sure? Since this is a Qt slot, we would need to write that const in all SIGNAL() and SLOT() macros.